core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -400,6 +406,47 @@ protected static XContentBuilder toXContent(QueryBuilder<?> query, XContentType
     }
 
     /**
+     * Randomize the name of a query builder. Note that this can change the meaning of the query builder. Like <code>match</code> can turn
+     * into <code>match_phrase</code>.
+     */
+    private XContentBuilder randomizeName(XContentBuilder builder, QueryBuilder<?> query) throws IOException {
```

Comment:
I like this test, however this won't catch any accidental or unknowing changes (like removing or changing the primary name for example) from any developer. I think in addition we should explicitely check the NAME/NAMES constants in each individual test and have some comments / errors that prevent any accidental changes. Like we do for enum constants in some tests already. We can add those in follow up PRs for all queries though.